File: 프로그래머스/lv3/43162. 네트워크/네트워크.swift
import Foundation

func solution(_ n:Int, _ computers:[[Int]]) -> Int {
    var answer = 0
    var visited : [Bool] = Array(repeating: false, count: n)
    
    func dfs(_ start : Int){
        
        visited[start] = true
        
        for comp in 0..<n {
            if !visited[comp] && computers[start][comp] > 0 {
                dfs(comp)
            }
            
        }
    }
    
    for i in 0..<n {
        if !visited[i]{
            answer += 1
            dfs(i)
        }
    }
    return answer
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
from collections import deque
visited = [False]
def solution(n, computers):
    global visited
    visited *= n
    answer = 0
    
    
    def dfs(idx):
        global visited
        visited[idx] = True
        for cmp in range(n):
            if not visited[cmp] and computers[idx][cmp] > 0:
                dfs(cmp)
    
    for i in range(n):
        if not visited[i]:
            answer += 1
            dfs(i)
    return answer